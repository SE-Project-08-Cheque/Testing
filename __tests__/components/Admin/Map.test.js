// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup, fireEvent, queryByTestId } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import Map from "../../../components/Admin/Map";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Map />, div);
});

it("renders map correctly", () => {
  const onClickonMap = (longitude, latitude) => {
    expect(longitude).toBeDefined();
    expect(latitude).toBeDefined();
  };
  const { getByTestId } = render(<Map onClickonMap={onClickonMap} />);

  fireEvent.click(getByTestId("map"));
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <Map/>
  );

  expect(tree).toMatchSnapshot();
});
