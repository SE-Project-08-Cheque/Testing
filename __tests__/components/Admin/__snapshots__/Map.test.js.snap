// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-0"
  data-testid="map"
>
  <div
    style={
      Object {
        "cursor": "grab",
        "height": 400,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "absolute",
            "visibility": "inherit",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
